diesel/src/mysql/connection/bind.rs

```diff
@@ -170,6 +136,33 @@ impl BindData {
         if let Some(ref mut is_truncated) = self.is_truncated {
             bind.error = is_truncated;
         }
+
+        bind
+    }
+
+    /// Resizes the byte buffer to fit the value of `self.length`, and returns
+    /// a tuple of a bind pointing at the truncated data, and the offset to use
+    /// in order to read the truncated data into it.
+    ///
+    /// This invalidates the bind previously returned by `mysql_bind`. Calling
+    /// this function is unsafe unless the binds are immediately rebound.
+    unsafe fn bind_for_truncated_data(&mut self) -> Option<(ffi::MYSQL_BIND, usize)> {
+        if self.is_truncated() {
+            let offset = self.bytes.capacity();
+            let truncated_amount = self.length as usize - offset;
+
+            debug_assert!(truncated_amount > 0, "output buffers were invalidated \
+                without calling `mysql_stmt_bind_result`");
```

Comment:
(This catches `truncated_amount == 0`, underflow already triggers an assert in debug)